tests/Doctrine/Tests/DBAL/Query/Expression/CompositeExpressionTest.php

```diff
@@ -77,4 +77,12 @@ public function provideDataForConvertToString()
             ),
         );
     }
+
+    public function testAddPreventsAddingEmptyCompositeExpressionObjects()
+    {
+        $emptyExpr = new CompositeExpression(CompositeExpression::TYPE_AND, []);
+        $expr = new CompositeExpression(CompositeExpression::TYPE_AND, [$emptyExpr]);
+
+        $this->assertEquals(0, count($expr));
```

Comment:
I don't have a strong opinion on this TBH and I don't know what the original intention was /cc @Ocramius @guilhermeblanco